pkg/api/types.go

```diff
@@ -744,13 +744,18 @@ type Service struct {
 	Status ServiceStatus `json:"status,omitempty"`
 }
 
+type EndpointEntry struct {
+	Endpoint        string `json:"endpoint"`
+	ObjectReference `json:"metadata,omitempty"`
```

Comment:
I was thinking that the object reference would be one field of a more general metadata map. 

Also, I want to reserve the term `metadata` for ObjectMeta/ListMeta.

What I proposed in #2585 was:
```
	Info map[string]string `json:"info,omitempty"`
```

I understand it would be more convenient for Kubernetes-specific applications for a first-class ObjectReference field. I have been concerned about introducing too many Kubernetes-isms into this API, but perhaps I should just accept that we'll need an adaptor to a more generic API, as @thockin recommended. In that case, I'd make the field name more specific, such as `targetRef`.